instance/models/openedx_instance.py

```diff
@@ -392,3 +396,63 @@ def delete(self, *args, **kwargs):
         self.deprovision_s3()
         self.deprovision_rabbitmq()
         super().delete(*args, **kwargs)
+
+    def update_consul_metadata(self):
+        """
+        This method is going over all fields in this model to reflect
+        their values on Consul.
+        Values on consul will be fetched first, then updated only if
+        they changed in this version of the model using agent's `cas`
+        parameter (Check-And-Set). If PUT operation successfully
+        updated the field then we update the configuration version
+        number in Consul.
+
+        :note: This still doesn't apply removed-configurations case.
+
+        :return: The current version number, and the if they include
+                 changes or not.
+        """
+        if not settings.CONSUL_ENABLED:
+            return 0, False
+
+        agent = consul.Consul()
+        active_servers_num = len(self.get_active_appservers())
+        consul_prefix = '{ocim}/instances/{instance}/'.format(
+            ocim=settings.OCIM_ID, instance=self.pk)
+
+        # Collect the current version's configurations
+        new_conf = {
+            'id': str(self.pk),
+            'ocim': settings.OCIM_ID,
+            'name': self.name,
+            'domains': str(self.get_load_balanced_domains()),
+            'health_checks_enabled': str(int(active_servers_num > 1)),
+            'basic_auth': self.http_auth_info_base64(),
+            'active_app_servers': str([
+                server.name for server in self.get_active_appservers()
+            ]),
+        }
+
+        # Reflect new configurations on Consul
+        includes_changes = True
+        _, consul_conf = agent.kv.get(consul_prefix, recurse=True)
+        consul_conf = {
+            conf['Key']: conf['Value']
+            for conf in consul_conf
+        } if consul_conf else {}
```

Comment:
I also liked the approach you originally had, using check-and-set, which may work without first reading all the data.  If you get that approach working, that would be great, too.